core/payment-driver/zksync/src/driver.rs

```diff
@@ -471,14 +471,31 @@ impl PaymentDriverCron for ZksyncDriver {
                     .collect();
 
                 if let Err(err) = tx_success {
-                    log::error!(
-                        "ZkSync transaction verification failed. tx_details={:?} error={}",
-                        tx,
-                        err
-                    );
-                    self.dao.transaction_failed(&tx.tx_id).await;
-                    for order_id in order_ids.iter() {
-                        self.dao.payment_failed(order_id).await;
+                    // In case of invalid nonce error we can retry sending transaction.
+                    // Reset payment and transaction state to 'not sent', so cron job will pickup
+                    // transaction again.
+                    if err.contains("nonce") {
```

Comment:
I'd say that the safest would be to check some error code, but not sure if there are error codes standardized.
To be sure, I'd recommend more strict check here
```suggestion
                    if err.contains("Nonce mismatch") {
```